Fix rmilter socket path, add deprecation notice

avnik opened the pull request:
###### Motivation for this change

Putting socket and pid file in same workdir cause problem, so I moved socket out of workdir.
Also as rmilter is no more recomended (use built-in rspamd milter instead), deprecation notice was added.

/cc @samueldr this commits suggested for release-18.09 branch

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

avnik committed:
rmilter: move rmilter.sock out of /run/rmilter

/run/rmilter is set by systemd, and have root:root ownership, which
prevent pid file to write.

This fix suggested to be promoted to 18.09 branch.
(Although rmilter itself is deprecated, and I plan to remove it, after
18.09 would be released)

avnik committed:
rmilter: deprecation notice

joachifm commented:
See also https://github.com/NixOS/nixpkgs/pull/46637

Mic92 commented on nixos/modules/services/mail/rmilter.nix:
```diff
@@ -193,6 +193,9 @@ in
 
     (mkIf cfg.enable {
+      warnings = [
+        ''`config.serivces.rmilter' is deprecated, `rmilter' deprecated and unsupported by upstream, and would be removed from next releases ''
```
`config.services.rmilter` and `will be removed`.

avnik commented:
On Sun, Sep 23, 2018 at 06:43:43AM -0700, Jörg Thalheim wrote:

> > @@ -193,6 +193,9 @@ in
>    config = mkMerge [
>  
>      (mkIf cfg.enable {
> +      warnings = [
> +        ''`config.serivces.rmilter' is deprecated, `rmilter' deprecated and unsupported by upstream, and would be removed from next releases ''
> 
> `config.services.rmilter` and `will be removed`.
> 

@Mic92 thank you. Fixed, and added suggestion to use rspamd builtin
milter as well.

Mic92 commented:
If you are going to remove rmilter before the 19.03 release, we should backport both commits right?

avnik commented:
On Sun, Sep 23, 2018 at 05:15:56PM -0700, Jörg Thalheim wrote:

> If you are going to remove rmilter before the 19.03 release, we should backport both commits right?
> 

Thats plan.

If you suggest a slower upgrade path, we can backport only first commit
(with socket path bugfix), announce deprecation in post 18.09, and
remove it right after 19.03 branch.

But rmilter itself was added only as requirement to operate rspamd (0.6
was unable operate as milter directly, but 0.7+ can, and rmilter can be
avoided). So I suggest to all users switch to builtin rspamd's milter.

Mic92 commented:
backport: 4307b377819 and 1a159fe86ba
